includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -991,6 +991,29 @@ protected function extract_response_body( $response ) {
 				$error_message
 			);
 
+			if ( 'wcpay_amount_too_low' === $error_code
+				&& isset( $response_body['data']['min_amount'], $response_body['data']['currency'] )
+			) {
+
+				$min_amount = $response_body['data']['min_amount'];
+				$currency   = strtolower( $response_body['data']['currency'] );
+
+				if ( ! in_array( $currency, WC_Payments_Utils::zero_decimal_currencies(), true ) ) {
+					$min_amount = $min_amount / 100;
+				}
+
+				// Response contains a minumum order amount which we will
+				// use to build a specific message for a client.
+				$message = sprintf(
+					// translators: %1$s is a formatted amount with currency code.
+					__(
+						'Sorry, the minimum allowed order total is %1$s to use this payment method.',
+						'woocommerce-payments'
+					),
+					wc_price( $min_amount, [ 'currency' => strtoupper( $currency ) ] )
```

Comment:
@marcinbot, my main reason to parse it here was that I pass some additional data from the server side, i.e. minimum required amount and (settlement) currency. Existing flows seem to discard that additional data.

On the other hand I agree, it might be better to extend `API_Exception` and make it carry additional data along. I'll take a look at it.

Reply:
Added that mentioned PR here: https://github.com/Automattic/woocommerce-payments/pull/1268

Reply:
I decided to go a similar way, but to make an exception class to return a properly formatted message instead of replacing it when notice is created.